internal/utils/url_test.go

```diff
@@ -180,3 +180,59 @@ func Test_GetURLWithScheme(t *testing.T) {
 		})
 	}
 }
+
+func Test_GenerateTenantURL(t *testing.T) {
+	tenantID := "abc"
+	testCases := []struct {
+		name, baseURL, tenantID, expectedURL string
+		expectedErr                          error
+	}{
+		{
+			name:        "returns the correct tenant URL",
+			baseURL:     "http://bluecorp.org.local",
+			tenantID:    tenantID,
+			expectedURL: fmt.Sprintf("http://%s.bluecorp.org.local", tenantID),
+			expectedErr: nil,
+		},
+		{
+			name:        "returns the correct tenant URL when it has port",
+			baseURL:     "http://bluecorp.org.local:3000",
+			tenantID:    tenantID,
+			expectedURL: fmt.Sprintf("http://%s.bluecorp.org.local:3000", tenantID),
+			expectedErr: nil,
+		},
+		{
+			name:        "returns the correct tenant URL when it has a path",
+			baseURL:     "http://bluecorp.org.local/sdp",
+			tenantID:    tenantID,
+			expectedURL: fmt.Sprintf("http://%s.bluecorp.org.local/sdp", tenantID),
+			expectedErr: nil,
+		},
+		{
+			name:        "returns error when invalid base URL - no protocol and URL separator",
+			baseURL:     "bluecorp.org.local:3000",
+			tenantID:    tenantID,
+			expectedURL: "",
+			expectedErr: fmt.Errorf("invalid base URL: bluecorp.org.local:3000"),
+		},
+		{
+			name:        "returns error when invalid base URL - no protocol",
+			baseURL:     "://bluecorp.org.local:3000",
+			tenantID:    tenantID,
+			expectedURL: "",
+			expectedErr: fmt.Errorf("invalid base URL: ://bluecorp.org.local:3000"),
+		},
+	}
```

Comment:
Nice tests! Maybe adding one with `https` to show that different protocols are handled.